### PR TITLE
New fungal zombies.

### PR DESCRIPTION
**Pull request by nexusmrsep**
#### Summary
 ```SUMMARY: Content "New fungal zombies"```

#### Purpose of change
Partially answers to #32254, by adding new thematic fungal zombies that can appear due to fungalization effects of existing fungi.

#### Describe the solution
This PR adds new fungal zombies:
- pollinator zombie, successor of smoker zombie, emits constant plumes of fungal haze, the idea here is that fungalization process utilizes the smoke emitting properties to propagate fine spore dust, like the one fungal blossoms, its not as spread out as the smoke, but it has powerful effect
- fungal juggernaut, successor of skeletal juggernaut, slow, blind version of it's former self, more like a husk then a real threat, but still tough, ~~can fungalize ground it walks on,~~
- bloated fungal zombie, successor of bloated zombie, it explodes with a cloud of fungal haze, idea behind it is that fungalization made this zombie like a balloon full of spores - it was bloated before, now just the contents changed.
- [edit] fungal spider that can fungalize ground it walks on (but with no more adverse effects)

Every new zombie shares the properties of it's kin: they are slower, have less hp, their senses are dulled. Mycus does not produce fierce predators, just hosts to carry spores further into the world. Their menace comes from the spores they carry and disperse, not from properties of hosts. In these cases the properties of the decedent zombies vary - those beneficial to the spread of fungi are utilized, those that are not - simply don't matter. This is why fungalization actually nerfs the juggernaut, while pollinator zombie and bloated zombies get adopted to the ways of Mycus.

#### Describe alternatives you've considered
There are few other zombie types that might be fungalized into something thematically interesting. But since I can't invent anything brilliant and interesting on the fly, it may wait for some inspiration.

#### Additional context
- I see potential in rewriting monster funagalization block, but it's not that important now.
- It would be also nice if all fungal creatures were to a separate json file, to unbloat the `monsters.json` and keep the fungi in one place.

### CODE REVIEW

File: src/mondeath.cpp [hunk 1]
```diff
@@ -705,6 +702,21 @@ void mdeath::smokeburst( monster &z )
...
+{
+    // If the fungus died from anti-fungal poison, don't pouf
+    if( g->m.get_field_intensity( z.pos(), fd_fungicidal_gas ) ) {
+        if( g->u.sees( z ) ) {
+            add_msg( m_good, _( "The %s inflates and melts away." ), z.name() );
+        }
+        return;
+    }
+
+    std::string explode = string_format( _( "a %s explode!" ), z.name() );
```
Comment (KorGgenT) [inline src/mondeath.cpp]:
```suggestion
    std::string explode = string_format( _( "a %s explodes!" ), z.name() );
```

**Comment by Night-Pryanik** [comment]
> can fungalize ground it walks on

Pollinator and bloated fungal zombies have some rationale on why they have their features and properties, while fungal juggernaut doesn't. How is he able to produce infinite materials to turn ground into fungal terrain on the fly? I feel its ability is a little... far-fetched. Ability for the sake of ability.

**Comment by nexusmrsep** [comment]
@Night-Pryanik 
Ok, I can agree to that. 

I wanted the fungalized "something" to leave a trail of fungal ground where it goes, but without any extra adverse effects. So have in mind that it's the mildest terraforming ability, it should only fungalize the ground, so no spores, no fungaloids popping out of nowhere etc.

Perhaps I was trying too hard to push it for the juggernaut. I can and will remove it from the juggernaut, but I'd like to leave the mechanic for some other fungalized zombie or other specie. Can anyone point out any type that could inherit this effect, co I can code it in in this PR?

**Comment by DemAvalon** [comment]
Zombie hollow but full of spores? that is the only one I can think of.

**Comment by NeviNovat** [comment]
> @Night-Pryanik
> Ok, I can agree to that.
> 
> I wanted the fungalized "something" to leave a trail of fungal ground where it goes, but without any extra adverse effects. So have in mind that it's the mildest terraforming ability, it should only fungalize the ground, so no spores, no fungaloids popping out of nowhere etc.
> 
> Perhaps I was trying too hard to push it for the juggernaut. I can and will remove it from the juggernaut, but I'd like to leave the mechanic for some other fungalized zombie or other specie. Can

Well if the fungus made the juggernaut blind and less powerful ( as specified in this PR ) you can say that pieces of fungus fall from it as it moves, from what I see the fungus only cares about spreading, not about having powerful creatures ( unless maybe you are near the source ) so I can see the fungus not caring for the complex structure of the skeletal juggernaut since it doesn't help in spreading.

I am ok with the ability of the fungal juggernaut, but then again I don't know a lot about the fungus and maybe I am wrong and it doesn't make sense or maybe it'll be annoying or look weird.

**Comment by FulcrumA** [comment]
>Can anyone point out any type that could inherit this effect, co I can code it in in this PR?

Acid ants drip acid. Maybe some infested acid ants could drip some spore-infested sludge onto the ground? I know it's not really zombies, but I wonder whether regular animals shouldn't be infectable as well, at least over period of time, as per some fluff. Barring that, I guess some acid zombie which doesn't really use acid but drips said spore sludge onto the ground would do.

**Comment by nexusmrsep** [comment]
How about a fungal spider dragging it's spore-ridden abdomen on the ground? 
image

**Comment by Night-Pryanik** [comment]
>How about a fungal spider dragging it's spore-ridden abdomen on the ground?

Cool!

**Pull request merged**